### PR TITLE
[Main Course] - Add Python dict post

### PR DESCRIPTION
**Pull request by Realvincentyuan**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (Add Python dict post)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Main_Course/Coding/2021-08-21-Python-Dict.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+---
+
+# Python列表常用操作小技巧
+
+!!! info
+    作者：袁子弹起飞，发布于2021-08-21，阅读时间：约6分钟，微信公众号文章链接：[:fontawesome-solid-link:](https://mp.weixin.qq.com/s?__biz=MzI4Mjk3NzgxOQ==&mid=2247484437&idx=1&sn=6d58dbd242157e216cb0e573678686d9&chksm=eb90f761dce77e7711ad25f26be3ff212d80db386a74d5902af1247e027e1108c84ec8d0fb7c&token=891223383&lang=zh_CN#rd)
+
+## 1 前言
+
+在上一篇文章中，我们回顾了[Python列表的常用操作](https://mp.weixin.qq.com/s?__biz=MzI4Mjk3NzgxOQ==&mid=2247484437&idx=1&sn=6d58dbd242157e216cb0e573678686d9&chksm=eb90f761dce77e7711ad25f26be3ff212d80db386a74d5902af1247e027e1108c84ec8d0fb7c&token=379425388&lang=zh_CN#rd)，列表作为一场常用的数据类型在日常工作中扮演了非常重要的作用，这篇文章我们继续聊聊Python里另一种常用的数据类型 - 字典（Dict）。
```
Comment (jhqian0909) [inline docs/Main_Course/Coding/2021-08-21-Python-Dict.md:12]: 一个常用的

File: docs/Main_Course/Coding/2021-08-21-Python-Dict.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+定义字典可以使用`dict()`方法，或者使用花括号`name2code = {'Tony':1, 'Kevin':2, 'Luis':3}`，如果想要增加元素，可以使用键值对的赋值模式：`name2code['Nick'] = 0`。很容易看出，不同于列表，字典并不以整数作为下标。接下来我们来看看字典常用的方法。
```
Comment (jhqian0909) [inline docs/Main_Course/Coding/2021-08-21-Python-Dict.md:14]: 这个问号是什么呀

File: docs/Main_Course/Coding/2021-08-21-Python-Dict.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+True
+```
+
+### 2.2 删除元素
+
+字典删除元素的方法与列表类似：
+
+- 清空字典：`name2code.clear()`，注意clear()方法没有返回值
+- 返回键k对应的值，然后移除该键值对：`name2code.pop(k, [default])`
+- 随机返回一个键值，并移除该键值对：`name2code.popitem()`
```
Comment (jhqian0909) [inline docs/Main_Course/Coding/2021-08-21-Python-Dict.md:63]: 是移除最后一个吗

File: docs/Main_Course/Coding/2021-08-21-Python-Dict.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+### 2.3 循环
+
+使用for循环可以遍历字典的键，请注意，因为键可散列，所以其出现不遵循特定的顺序，以下代码在你的电脑上运行可能会有不同的输出顺序：
```
Comment (jhqian0909) [inline docs/Main_Course/Coding/2021-08-21-Python-Dict.md:67]: 我试竟然是不随机的

Comment (jhqian0909) [reply docs/Main_Course/Coding/2021-08-21-Python-Dict.md:67]: 可能是python版本的问题，我看说3.6之后dict默认是有序的

File: docs/Main_Course/Coding/2021-08-21-Python-Dict.md (hunk 1 above)
Comment (Realvincentyuan) [reply docs/Main_Course/Coding/2021-08-21-Python-Dict.md]: 已修改

File: docs/Main_Course/Coding/2021-08-21-Python-Dict.md (hunk 2 above)
Comment (Realvincentyuan) [reply docs/Main_Course/Coding/2021-08-21-Python-Dict.md]: 好像出现了乱码，本地没有，已经删除

File: docs/Main_Course/Coding/2021-08-21-Python-Dict.md (hunk 3 above)
Comment (Realvincentyuan) [reply docs/Main_Course/Coding/2021-08-21-Python-Dict.md]: > 是移除最后一个吗

后加进来的，先pop出去，语句已经修改，让语义更清楚

File: docs/Main_Course/Coding/2021-08-21-Python-Dict.md (hunk 4 above)
Comment (Realvincentyuan) [reply docs/Main_Course/Coding/2021-08-21-Python-Dict.md]: 我也查到了相关的说明，所以添加了一条注释，关于顺序的，谢谢Void！

**Review by jz263 — APPROVED** [review]
LGTM

**Pull request merged**